javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,138 @@
+//Item array removal
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+  ]; 
+  const nameToRemove = "Ahmad";
+  for(let arrayIndex of names ){
+    if(nameToRemove === arrayIndex){
+        names.splice(arrayIndex ,1)    
+    }
+  }
+  console.log('Remaining array after removel:  '+names);
+
+  //When will we be there??
+  const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+  };
+  function notThisFunctionName(){
+    return( travelInformation.destinationDistance/travelInformation.speed);
+  }
+  const travelTime = notThisFunctionName(travelInformation);
+  const timeinHhMm=timeConvertiontoHhMm( travelTime);
+  //refrece:how to covert decimal time to HH:mm  https://stackoverflow.com/questions/35460303/how-to-convert-decimal-hour-value-to-hhmmss/35460620
+  function timeConvertiontoHhMm(decimaltime){
+  let hrs = parseInt(Number(decimaltime));
+  let min = Math.round((Number(decimaltime)-hrs) * 60);
+  return(hrs+'hh:'+min+'mm');
+}
+  console.log('travel time duration => '+timeinHhMm); 
+
+
+  //Series duration of my life
+  const seriesDurations = [
+    {
+      title: "Game of thrones",
+      days: 3,
+      hours: 1,
+      minutes: 0,
+    },
+    {
+      title: "Sopranos",
+      days: 3,
+      hours: 14,
+      minutes: 0,
+    },
+    {
+      title: "The Wire",
+      days: 2,
+      hours: 12,
+      minutes: 0,
+    },
+  ];
+const lifeSpanInYear = 80;
+let sumofDurations = 0;
+percentageOfEachSerials(seriesDurations);
+  function percentageOfEachSerials(durations){
+      for(let idexofSreies of durations ){
+        let durationInHr = idexofSreies.hours+(idexofSreies.minutes/60);
+        let days = idexofSreies.days + (durationInHr/24);
+        let percentageOfEachSeries = ((days/365)/lifeSpanInYear)*100;
+        console.log(idexofSreies.title +' took '+percentageOfEachSeries.toFixed(3)+'% of my life');
+        sumofDurations = sumofDurations+percentageOfEachSeries;   
+      }
+      let percentageofsum = (sumofDurations/3).toFixed(3);
+      console.log('In total that is '+percentageofsum +'% of my life');
+  }
+
+
+//Smart-ease - Back to the basics!
+const notes = [];
+
+function saveNote(content, id) {
+  const addingNotes={content:content,id:id}
+  notes.push(addingNotes);
+}
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+console.log(notes);
+
+function getNote(id) {
+
+    for (const checkId of notes) {
+        if (checkId.id===id) {
+            let details = checkId;
+            return details;
+        }  
+    }
```

Comment:
`checkId` is a confusing name.

```suggestion
    for (const note of notes) {
        if (note.id===id) {
            let details = note;
            return details;
        }  
    }
```

Reply:
thank you for your good feedbacks
I have updated the problem based on the suggestions.